appinventor/appengine/src/com/google/appinventor/client/editor/simple/palette/SimpleComponentDescriptor.java

```diff
@@ -222,6 +247,31 @@ public SimpleComponentDescriptor(String name,
     COMPONENT_DATABASE = SimpleComponentDatabase.getInstance(editor.getProjectId());
   }
 
+  public SimpleComponentDescriptor(String name,
+                                   int version,
+                                   String versionName,
+                                   String dateBuilt,
+                                   String helpString,
+                                   String helpUrl,
+                                   String categoryDocUrlString,
+                                   boolean showOnPalette,
+                                   boolean nonVisible,
+                                   boolean external) {
+    // TODO: Remove editor and component database attributes
+    this.name = name;
+    this.editor = (YaFormEditor)Ode.getInstance().getCurrentFileEditor();
```

Comment:
This cast seems unnecessary as the field is a `SimpleEditor`. Also, this constructor is not called from anywhere so we could just remove it (and the YaFormEditor import above).